### PR TITLE
feat: Integrate TSTyche for Modern TypeScript Type Testing

### LINKED ISSUE #215: tstyche Typescript Tests
try [tsytche](https://tstyche.org/)

### PR DESCRIPTION
**Pull request by umxr**
### 📋 Summary

This PR modernises the type testing infrastructure by migrating from TSD to TSTyche, providing a more robust and maintainable type testing experience with better organisation and clearer output.

#### Package Configuration
- **Added `test:types` script** using TSTyche to `package.json`

#### Type Test Files
- **Created `types/core.tst.ts`** - Comprehensive test suite for core types
  - 22 tests covering `encodeCursor`, `decodeCursor`, `ConnectionArgs`, `PAGINATION_MODE`, `ConnectionResolverResponse`, and `connectionDirective`
  - 25 assertions validating function signatures, type constraints, and generic behaviors
- **Created `types/utils.tst.ts`** - Complete test coverage for utility types  
  - 14 test suites covering the `NodesOnly<T>` utility type
  - 15 assertions testing nested structures, generic constraints, and edge cases

#### Migration Benefits
- **Modern syntax**: Replaced TSD's `expectType`/`expectAssignable` with TSTyche's `expect().type.toBe<>()`/`expect().type.toBeAssignableTo<>()`
- **Better organization**: Tests are now grouped with `describe()` blocks and descriptive test names
- **Clearer output**: More readable test results with structured reporting
- **Enhanced debugging**: Better error messages and test isolation

### 🧪 Test Coverage

**Type Tests**: 36 tests, 40 assertions - All passing ✅
- Core type validation (encodeCursor, decodeCursor, ConnectionArgs, etc.)
- Enum and literal type checking (PAGINATION_MODE)
- Generic type constraints and transformations
- Complex nested resolver structures
- Optional fields and readonly arrays
- Union types and constraint validation

**Unit Tests**: 33 tests - All passing ✅  
**Code Coverage**: 100% maintained ✅

### 🔧 Usage

```bash
# Run type tests only
npm run test:types

# Run all tests (unit + types)
npm test

### 📝 Migration Notes

- All original TSD test assertions have been preserved and converted
- Test coverage remains comprehensive with no regression
- File naming follows TSTyche conventions (`.tst.ts` extension)
- Proper ES module imports with `.js` extensions for compatibility

### 🎯 Testing

- ✅ All type tests pass with TSTyche
- ✅ All unit tests continue to pass  
- ✅ Integration with existing CI/CD pipeline
- ✅ No breaking changes to public API
- ✅ Backward compatibility maintained

This migration establishes a solid foundation for future type testing and ensures our GraphQL pagination directive types remain robust and well-validated.

closes #215

### CODE REVIEW

File: types/core.tst.ts [hunk 1]
```diff
@@ -0,0 +1,260 @@
+import { expect, test, describe } from 'tstyche'
+import {
+  encodeCursor,
+  decodeCursor,
+  PAGINATION_MODE,
+  connectionDirective
+} from './core.js'
+import type { ConnectionArgs, ConnectionResolverResponse } from './core.js'
```
Comment (mrazauskas) [inline types/core.tst.ts:8]: Since this PR is an experiment, would you mind adding the following to `package.json`:

```json
"exports": {
  ".": "./index.js"
},
```

After that you can rewrite these imports: `import xyz from 'relay-pagination-directive'`. This way the root `index.d.ts` will be used an issues with it will be revealed. For instance, it should not have extension less imports.

---

The change in `package.json` allows using self-reference import. See Node.js documentation for more details: https://nodejs.org/docs/latest/api/packages.html#self-referencing-a-package-using-its-name

By the way, you can use `import xyz from 'relay-pagination-directive'` in unit tests as well.

File: types/core.tst.ts [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+describe('ConnectionArgs type', () => {
+  test('should accept first and after properties', () => {
+    expect({
+      first: 10,
+      after: 'cursor123'
+    }).type.toBeAssignableTo<ConnectionArgs>()
+  })
+
+  test('should accept only first property', () => {
+    expect({ first: 5 }).type.toBeAssignableTo<ConnectionArgs>()
```
Comment (mrazauskas) [inline types/core.tst.ts:35]: Here an in similar cases better use `.toBeAssignableWith()`:

```suggestion
    expect<ConnectionArgs>().type.toBeAssignableWith({ first: 5 })
```

This way the value under test will always stay on the left hand side. Easier to read. Otherwise it does the same job.

**Review by simoneb — APPROVED** [review]
